api/views.py

```diff
@@ -15,3 +23,30 @@ class AuthorViewSet(viewsets.ModelViewSet):
     serializer_class = AuthorSerializer
     permission_classes = [permissions.IsAuthenticated]
     http_method_names = ['get']
+
+
+class PostViewSet(viewsets.ModelViewSet):
+    renderer_classes = [JSONRenderer]
+    pagination_class = page_number_pagination_class_factory([('type', 'posts')])
+
+    # TODO: set up the posts model
+    queryset = Post.objects.order_by('id')
+    serializer_class = PostSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    http_method_names = ['get']
+
+    # detail indicates  whether we can do this on the list (false), or only a single item (true)
+    @action(methods=['get'], detail=True, url_path='image', name='image')
+    def image(self, request, **kwargs):
+        author_id = kwargs['author_pk']
+        post_id = kwargs['pk']
+
+        img = get_object_or_404(Post.objects, author_id=author_id, pk=post_id)
+
+        if img.content_type != ContentType.PNG and img.content_type != ContentType.JPG:
+            return Response(status=404)
+
+        with open(os.path.abspath(settings.BASE_DIR) + img.img_content.url, 'rb') as img_file:
+            encoded_img = base64.b64encode(img_file.read()).decode('utf-8')
+
+        return HttpResponse(f'data:{img.content_type},{encoded_img}', content_type=img.content_type)
```

Comment:
Do you think it might be better to just return the base64 encoded (without the data prefix) since they already know the post type (from contents) and will likely add their own prefix?

Reply:
This was definitely my biggest question when I was working on this, as I found that the spec isn't very clear on this. Originally I was just returning the encoded image. It would be simple enough for an end user to add their own "data:{content_type},{base64}" so I will amend it to that.